### ISSUE TITLE
Failover with Elasticache cluster causes excessive failures, RedisNodeNotFoundException

### ISSUE THREAD

**Issue by JLP-in-SD** [redisson/redisson#2707]
**Expected behavior**
Elasticache Redis cluster failover on primary should not cause several minutes of failing Redis calls.

**Actual behavior**
After failover ClosedChannelException caused by WriteRedisConnectionException happen.
Requests timeout and fail with WriteRedisConnectionException.

Note: this looks like the same problem as issue #2678 

**Steps to reproduce or test case**
Run application
Do "failover primary" on a shard from AWS Console

**Redis version**
5.0.4

**Redisson version**
3.12.4

**Redisson configuration**
```
clusterServersConfig:
  readMode: "MASTER"
  nodeAddresses:
  - "rediss://clustercfg.redis-cluster.aaa.usw2.cache.amazonaws.com:6379"
  tcpNoDelay: "true"
```

**Comment by JLP-in-SD** [comment]
[JLP_test_run_200414.zip](https://github.com/redisson/redisson/files/4495710/JLP_test_run_200414.zip)

**Comment by mrniko** [comment]
> ClosedChannelException caused by WriteRedisConnectionException happen

You need to increase `retryAttempts` or `retryInterval` parameters:

https://github.com/redisson/redisson/wiki/2.-Configuration#retryattempts
https://github.com/redisson/redisson/wiki/2.-Configuration#retryinterval

**Comment by JLP-in-SD** [comment]
@mrniko
Wouldn't I want to *decrease* the `retryInterval` value, so it can retry more times in a shorter time?

What are some starting values that you would recommend?

**Comment by JLP-in-SD** [comment]
@mrniko Ok. I increased `retryAttempts` * `retryInterval` to be long enough to get past failover (30 and 1000 respectively).
It worked. Thanks.

**Comment by kschroiff** [comment]
We just had 2 incidents on AWS based on 100 retries, retry interval 2000ms and scan interval 2000ms ...
This didn't help - we had to restart all app servers.
In my book there's a more fundamental issue somewhere.

**Comment by mrniko** [comment]
@kschroiff 

Could you share exception stacktrace and version you use?

**Comment by kschroiff** [comment]
After some further investigations today it seems as if our issue was related to this ticket here:
* https://github.com/redisson/redisson/issues/2234

We were running on 3.10.4. 

Given that the issue was fixed in 3.11.2 (I hope), we'll now upgrade to the latest version.
